updater/internal/service/service_linux.go

```diff
@@ -72,20 +72,38 @@ type linuxService struct {
 
 // Start the service
 func (l linuxService) Start() error {
-	//#nosec G204 -- serviceName is not determined by user input
-	cmd := exec.Command("systemctl", "start", l.serviceName)
-	if err := cmd.Run(); err != nil {
-		return fmt.Errorf("running systemctl failed: %w", err)
+	var cmd *exec.Cmd
+	if l.serviceCmdName == "service" {
```

Comment:
I think I'd rather we take an approach here of having to implementations of the Service interface (one for systemd, one for sysv) instead of having these if statements; I think it would be more straight-forward and reduce the number of redundant checks here.

Reply:
This is now done, thank you for the advice.